cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -44,7 +44,7 @@ import (
 const (
 	DefaultCloudConfigPath = "/etc/kubernetes/cloud-config"
 
-	defaultv17AdmissionControl = "Initializers,NamespaceLifecycle,LimitRanger,ServiceAccount,PersistentVolumeLabel,DefaultStorageClass,DefaultTolerationSeconds,NodeRestriction,ResourceQuota"
+	defaultv17AdmissionControl = "Initializers,NamespaceLifecycle,LimitRanger,ServiceAccount,PersistentVolumeLabel,DefaultStorageClass,DefaultTolerationSeconds,LimitPodHardAntiAffinityTopology,NodeRestriction,ResourceQuota"
```

Comment:
This will not work if kubeadm deploys a v1.7 cluster, which must be supported.
You must make two copies of this list:
 - `defaultv17AdmissionControl` keep as-is
 - `defaultv18AdmissionControl` add this change

Add a constant in `cmd/kubeadm/app/constants/constants.go`: `MinimumLimitPodHardAntiAffinityTopologyVersion` or something like that and set to `v1.8.0-alpha.3` (assuming this is available in the coming `alpha.3`)

Finally, add a conditional in `getAPIServerCommand` that sets `defaultArguments["admission-control"] = defaultv18AdmissionControl` if current `k8sVersion.AtLeast(kubeadmconstants.MinimumLimitPodHardAntiAffinityTopologyVersion)`

Sounds good?
Also make the corresponding unit tests please (add new test case for this)